### DISCUSSION TITLE
Alias in subquery and the use in the same subqueries filter

### DISCUSSION THREAD

**Discussion by C-monC** [sqlalchemy/sqlalchemy#11020] [Usage Questions]
Hi,

I am really struggling to get this right.
This is the sql I am trying to reproduce:
```
.....  AND (
        SELECT COUNT(*)
        FROM jsonb_array_elements(edm.doc_metadata) AS elem
        WHERE (elem->>'claim') ILIKE '%bulk%'
      ) >= 2;
```

What's critical here is that the from clause is named.
Doing this results in "Set returning function can not be used in where clause"
  AND (
        SELECT COUNT(*)
        FROM jsonb_array_elements(edm.doc_metadata) AS elem
        WHERE (jsonb_array_elements(edm.doc_metadata)->>'claim') ILIKE '%bulk%'
      ) >= 2;
      
This is the closest I have come to getting the subquery right:

                    subquery = (
                        select(func.count())
                        .select_from(
                            func.jsonb_array_elements(
                                tbl.__table__.columns[var_record[1]]
                            )
                        )
                        .where(
                            func.jsonb_array_elements(
                                tbl.__table__.columns[var_record[1]]
                            )
                            .op("->>")(var_record[3])
                            .ilike(var_values[0])
                        )
                    )

Is it possible to assign a label/alias to a table and then use it in the same query?

If I try alias it:
```
                    flattened_array = func.jsonb_array_elements(
                        tbl.__table__.columns[var_record[1]]
                    ).alias()
                    subquery = (
                        select(func.count())
                        .select_from(flattened_array)
                        .where(
                            flattened_array.columns[0].op("->>")(var_record[3]).ilike(
                                var_values[0]
                            )
                        )
                    )
```

Then the output query looks like this:

```
SELECT count(*) AS count_1 
FROM jsonb_array_elements(embedded_document_metadata.doc_metadata) AS anon_1 
WHERE lower((anon_1.jsonb_array_elements_1 ->> :param_1)) LIKE lower(:param_2) 
```

**Comment by CaselIT** [comment]
Hi,

You can look at
- https://docs.sqlalchemy.org/en/20/core/functions.html#sqlalchemy.sql.functions.FunctionElement.table_valued
- https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.TableValuedAlias.render_derived
- https://docs.sqlalchemy.org/en/20/core/functions.html#sqlalchemy.sql.functions.FunctionElement.column_valued
